rs/nns/governance/src/storage/neurons.rs

```diff
@@ -230,6 +230,58 @@ where
         Ok(self.reconstitute_neuron(neuron_id, main_neuron_part, sections))
     }
 
+    pub fn register_recent_neuron_ballot(
+        &mut self,
+        neuron_id: NeuronId,
+        topic: Topic,
+        proposal_id: ProposalId,
+        vote: Vote,
+    ) -> Result<(), NeuronStoreError> {
+        if topic == Topic::ExchangeRate {
+            return Ok(());
+        }
+
+        let main_neuron_part = self
+            .main
+            .get(&neuron_id)
+            // Deal with no entry by blaming it on the caller.
+            .ok_or_else(|| NeuronStoreError::not_found(neuron_id))?;
+
+        let recent_entry_index = main_neuron_part.recent_ballots_next_entry_index;
+
+        let next_entry_index = if let Some(recent_entry_index) = recent_entry_index {
+            recent_entry_index as usize
+        } else {
+            let mut ballots = read_repeated_field(neuron_id, &self.recent_ballots_map);
+            ballots.reverse();
+            let next_entry = ballots.len() % MAX_NEURON_RECENT_BALLOTS;
+            update_repeated_field(neuron_id, ballots, &mut self.recent_ballots_map);
+            next_entry
+        };
```

Comment:
Again, putting the reversal logic and setting the next index in an `ensure_...` function would be more clear. (I recognize this is a separate case from the above.)

Reply:
I don't see how this would help in this case, since main_neuron_part and then the recent_ballots_map and then the neuron_id would all be parameters, which would make it a much clunkier function.  In the other case, it could be nicely encapsulated, but here I don't think it would be nicer.